src/main/java/Duke.java

```diff
@@ -1,10 +1,47 @@
+import java.io.File;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Scanner;
+
+/**
+ * Represents the personal assistant and contains the main method.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private TaskList list;
+    private Ui ui;
+    private Parser parser;
+    private Storage storage;
+
+    /**
+     * Creates a duke object which stores data in the specified path .
+     * @param s the absolute path of the file where the data is stored.
+     */
+    public Duke(String s) {
+        list = new TaskList();
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(s);
+    }
+
+    /**
+     * Contains the flow of the program execution.
+     * @throws ParseException if date of event class not in specified format.
+     * @throws IOException if file not found or other input output exceptions
+     */
+    public void run() throws ParseException, IOException {
+        Scanner scan = new Scanner(System.in);
+        this.ui.greet();
+        File f = new File("../Duke/data/Duke.txt");
+        Scanner sca = new Scanner(f);
```

Comment:
The variable names are a little confusing, perhaps you can be more explicit about the functionality of each by naming them `userInputScanner` and `fileScanner`.